R/vars.R

```diff
@@ -81,7 +81,17 @@ poke_vars <- function(vars) {
 #' @rdname poke_vars
 #' @export
 peek_vars <- function() {
-  vars_env$selected %||% abort("No tidyselect variables were registered")
+  if (is.null(vars_env$selected)) {
+    the_call <- sys.call(sys.parent()) 
+    fun <- as.character(the_call)[1]
```

Comment:
I think this would be a bit more flexible if `peek_vars()` took `fn = NULL` parameter. Then the caller could override the function name.

Also for robustness it would be nice to check if the call head (e.g. `the_call[[1]]` is a symbol. If it is not, we'd fall back to the default error message.

Reply:
Thank you! I was having trouble conceiving of what the argument (`fn = NULL`) should be. 

For the robustness aspect, something like this?

```r
if (is.symbol(the_call[[1]]) && !identical(the_call[[1]], as.symbol('peek_vars'))) {
  # call-specific
} else {
  abort("No tidyselect variables were registered")
}
```

Reply:
yup though I'm not sure we need to have a special-case for `peek_vars()`?

Reply:
If we didn't have the special case for peek_vars(), then the user would get the error "peek_vars() must be used within a *selecting* function." if they attempted to use it in the Global Environment, which doesn't make too much sense, since peek_vars() is not used in the selecting functions.

Reply:
Although, it does create its own host of issues for testing -_-

Reply:
hmm, though that's not entirely wrong either :) For simplicity I would just omit the special-casing.

The new `fn` parameter looks good!

Reply:
gotcha! Will fix.